### DISCUSSION TITLE
To synchronize amino substitution on different chain

### DISCUSSION THREAD

**Discussion by Venthaul** [RosettaCommons/rosetta#250] [Rosetta]
Hi!

I'm really new to Rosetta and I'm trying to do such a task where my target protein is a homologous trimer, but one of the chains A has a different conformation than the other two B and C. I want to mutate proteins in such a way that for the same site on different chains, say 300, amino acid substitution occurs synchronously, i.e. they become alanine or histidine at the same time. The goal of my subsequent analysis is to try to find mutations that make this protein more stable (or nearly so).

Can such a scanning protocol be implemented?

Thank you!

**Comment by everyday847** [comment]
It depends on how precise "at the same time" is. If the complex cannot treated with the actual symmetry machinery, your best bet may be an object like `LinkResidues`.

**Reply by Venthaul** [reply]
Thank you! I checked the discrition of ```LinkResidues``` and it looks like what I need.

**Reply by Venthaul** [reply]
Based on this page (https://docs.rosettacommons.org/docs/latest/scripting_documentation/RosettaScripts/TaskOperations/taskoperations_pages/LinkResidues), I wrote the following XML file for testing:

```
<ROSETTASCRIPTS>
    <SCOREFXNS>
        <ScoreFunction name="ref2015" weights="ref2015" />
    </SCOREFXNS>

    <TASKOPERATIONS>
        <ReadResfile name="resfile" filename="mutations.res" />
        <RestrictToRepacking name="rtr" />
        <LinkResidues name="linkres">
            <LinkGroup group="1,253"/>
            <LinkGroup group="5,257"/>
            <LinkGroup group="9,261"/>
            <LinkGroup group="13,265"/>
        </LinkResidues>
        <InitializeFromCommandline name="ifcl" />
        <IncludeCurrent name="ic" />
    </TASKOPERATIONS>

    <MOVERS>
        <PackRotamersMover name="pack" scorefxn="ref2015" task_operations="resfile,linkres,ifcl" />
        <FastRelax name="relax" scorefxn="ref2015" repeats="5" task_operations="ifcl,ic,rtr" />
    </MOVERS>

    <PROTOCOLS>
        <Add mover="pack" />
        <Add mover="relax" />
    </PROTOCOLS>
</ROSETTASCRIPTS>
```
Meanwhile, the ```mutations.res``` is:
```
NATAA
START
1 A ALLAA
5 A ALLAA
9 A ALLAA
13 A ALLAA
253 B ALLAA
257 B ALLAA
261 B ALLAA
265 B ALLAA
```
Segmentation fault (core dumping) occurs when run using the following methods:
```
rosetta_scripts.mpi.linuxgccrelease 
-in:file:s 3UKM_AB.pdb 
-out:suffix _design 
-ex1
-ex2
-use_input_sc true
-nstruct 100
```

When ```<PackRotamersMover name="pack" scorefxn="ref2015" task_operations="resfile,linkres,ifcl" />``` was changed to ```<PackRotamersMover name="pack" scorefxn="ref2015" task_operations="resfile,ifcl" />```, there was no error but I am not sure if the sites are really linked.

I also tried to use the totally same XML provided in ```LinkResidues``` and it also return Segmentation fault (core dumping).

Could you give me some advice or explanation on this? Thank you!
